zng/resolver/context.go

```diff
@@ -286,13 +286,42 @@ func (c *Context) Lookup(td int) *zng.TypeRecord {
 	return nil
 }
 
-func (c *Context) addTypeWithLock(typ zng.Type) {
-	key := typ.String()
+func setKey(inner zng.Type) string {
```

Comment:
This bug pre-dates this PR right? Especially if so, is there a way to test the bug fixed by this commit?

Reply:
I was working on that :-)

Reply:
done